src/client/datascience/jupyter/jupyterExecution.ts

```diff
@@ -192,13 +192,16 @@ export class JupyterExecutionBase implements IJupyterExecution {
                             IJupyterSessionManagerFactory
                         );
                         const sessionManager = await sessionManagerFactory.create(connection);
-                        kernelConnectionMetadata = await this.kernelSelector.getPreferredKernelForRemoteConnection(
-                            undefined,
-                            sessionManager,
-                            options?.metadata,
-                            cancelToken
-                        );
-                        await sessionManager.dispose();
+                        try {
+                            kernelConnectionMetadata = await this.kernelSelector.getPreferredKernelForRemoteConnection(
+                                undefined,
+                                sessionManager,
+                                options?.metadata,
+                                cancelToken
+                            );
+                        } finally {
+                            await sessionManager.dispose();
```

Comment:
An unrelated clean up, not required, but harmless & slightly better (again unrelated, couldn't help my self when fixing the typo). 😄